src/Core/Domain/Product/Command/UpdateProductOptionsCommand.php

```diff
@@ -78,12 +79,19 @@ class UpdateProductOptionsCommand
      */
     private $manufacturerId;
 
+    /**
+     * @var ShopConstraint
+     */
+    private $shopConstraint;
+
     /**
      * @param int $productId
+     * @param ShopConstraint $shopConstraint
      */
-    public function __construct(int $productId)
+    public function __construct(int $productId, ShopConstraint $shopConstraint)
```

Comment:
Isn't it a BC Break ?

Reply:
Yes and no, it's part of the development of product page and is mandatory to handle multishop correctly Since all this code was experimental we don't consider it as a BC break (like all other code related to new product page, but sometimes we added the BC break label, we're gonna have to clean them all).